### ISSUE TITLE
Handle fetching lots of file content in searcher-zoekt client

### ISSUE THREAD

**Issue by rvantonder** [sourcegraph/sourcegraph#17494]
In v1 of the searcher-zoekt client #17495, we currently hardcode a limit of `30` for the number of matches of something we are trying to find, and return file contents for any files containing matches. This to guard against OOM issues in Zoekt/Searcher because we currently have to execute a batch request for results from Zoekt, and store them on disk in a zip in searcher.

We need to figure out how we can run a search with a user set limit of `N`. The preferred solution would be to stream results from Zoekt and write to disk initially (other than that, I'm not sure if we can batch requests, since we can't paginate). 

A follow up change is to remove the restriction on writing to disk/zip, and just stream content directly to comby workers. But the disk-writing/zip files is an inefficiency and not a danger to OOM issues, so this is separate and lower priority.

**Comment by camdencheek** [comment]
It doesn't look like the RPC framework we're using with Zoekt supports any kind of streaming. I was thinking the current changes to add streaming results to the Zoekt interface would help here, but it appears that we just stream the results from requests split across multiple Zoekt endpoints rather than streaming matches from a single request. 

@keegancsmith Can you clarify if the plans for streaming from Zoekt are limited to streaming the result set from each endpoint, and won't be extended to streaming individual matches from a single endpoint?

If that's the case, we will still hit memory issues whenever the results from a single shard are large since we can't start writing to disk until the entire request is decoded.

**Comment by keegancsmith** [comment]
> @keegancsmith Can you clarify if the plans for streaming from 
> Zoekt are limited to streaming the result set from each 
> endpoint, and won't be extended to streaming individual matches 
> from a single endpoint?

We intend on making Zoekt have a streaming RPC interface. Likely 
will work on that next week.

**Comment by rvantonder** [comment]
Is there a tracking issue for ^ @keegancsmith ?

**Comment by keegancsmith** [comment]
#18303

**Comment by camdencheek** [comment]
Closing since we now stream Zoekt results directly to Comby.
